src/sampleutil.cpp

```diff
@@ -178,8 +178,14 @@ void SampleUtil::copyWithRampingGain(CSAMPLE* pDest, const CSAMPLE* pSrc,
 // static
 void SampleUtil::convertS16ToFloat32(CSAMPLE* pDest, const SAMPLE* pSrc,
         unsigned int iNumSamples) {
+    // Sample min can be -32768 instead of -32767, so make sure our conversion
+    // scale is picking the largest possible absolute value.
+    // Note that this means that although some sample values convert to -1.0,
+    // none will convert to +1.0.
+    static const CSAMPLE kConversionFactor = math_max(CSAMPLE(-SAMPLE_MIN),
```

Comment:
i think static might work against us here since it prevents doing the max at compile time:

```c++
#include <algorithm>
#include <limits.h>

float convert(float in) {
    const float convert = std::max(SHRT_MAX, -SHRT_MIN);
    return convert / in;
}
```

```asm
	.section	__TEXT,__text,regular,pure_instructions
	.section	__TEXT,__literal4,4byte_literals
	.align	2
LCPI0_0:
	.long	1191182336              ## float 32768
	.section	__TEXT,__text,regular,pure_instructions
	.globl	__Z7convertf
	.align	4, 0x90
__Z7convertf:                           ## @_Z7convertf
	.cfi_startproc
## BB#0:
	pushq	%rbp
Ltmp2:
	.cfi_def_cfa_offset 16
Ltmp3:
	.cfi_offset %rbp, -16
	movq	%rsp, %rbp
Ltmp4:
	.cfi_def_cfa_register %rbp
	vmovss	LCPI0_0(%rip), %xmm1
	vdivss	%xmm0, %xmm1, %xmm0
	popq	%rbp
	retq
	.cfi_endproc


.subsections_via_symbols
```

```c++
#include <algorithm>
#include <limits.h>

float convert(float in) {
    static const float convert = std::max(SHRT_MAX, -SHRT_MIN);
    return convert / in;
}
```

```asm

	.section	__TEXT,__text,regular,pure_instructions
	.globl	__Z7convertf
	.align	4, 0x90
__Z7convertf:                           ## @_Z7convertf
	.cfi_startproc
## BB#0:
	pushq	%rbp
Ltmp2:
	.cfi_def_cfa_offset 16
Ltmp3:
	.cfi_offset %rbp, -16
	movq	%rsp, %rbp
Ltmp4:
	.cfi_def_cfa_register %rbp
	subq	$16, %rsp
	movb	__ZGVZ7convertfE7convert(%rip), %al
	testb	%al, %al
	jne	LBB0_3
## BB#1:
	leaq	__ZGVZ7convertfE7convert(%rip), %rdi
	vmovss	%xmm0, -4(%rbp)         ## 4-byte Spill
	callq	___cxa_guard_acquire
	vmovss	-4(%rbp), %xmm0         ## 4-byte Reload
	testl	%eax, %eax
	je	LBB0_3
## BB#2:
	movl	$1191182336, __ZZ7convertfE7convert(%rip) ## imm = 0x47000000
	leaq	__ZGVZ7convertfE7convert(%rip), %rdi
	callq	___cxa_guard_release
	vmovss	-4(%rbp), %xmm0         ## 4-byte Reload
LBB0_3:
	vmovss	__ZZ7convertfE7convert(%rip), %xmm1
	vdivss	%xmm0, %xmm1, %xmm0
	addq	$16, %rsp
	popq	%rbp
	retq
	.cfi_endproc

.zerofill __DATA,__bss,__ZZ7convertfE7convert,4,2 ## @_ZZ7convertfE7convert
.zerofill __DATA,__bss,__ZGVZ7convertfE7convert,8,3 ## @_ZGVZ7convertfE7convert

.subsections_via_symbols
```